sign/eddsa/eddsa_test.go

```diff
@@ -130,6 +130,28 @@ func TestEdDSASigningMalleability(t *testing.T) {
 	assert.Error(t, Verify(ed.Public, msg, sig))
 }
 
+// Test non-canonical keys
+func TestEdDSASigningNonCanonical(t *testing.T) {
+	var nonCanonicalPk []byte = []byte{0xf6, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
```

Comment:
What is that number? Why is it noncanonical? Can you write a loop that creates canonical and non-canonical public keys for tests?

Also, if I understand correctly, this test will fail even with a canonical public key, no?

What I proposed was to write a test that directly tests the methods I listed, without using the `Sign` method.

Reply:
That number is a non-canonical point to be used as a public key in that function: the Verify function must fail with non-canonical public keys.

These tests are verifying the integrity of the edwards25519.Verify function, now reinforced with new functions (Ge25519IsCanonical, Sc25519IsCanonical...)

Reply:
How did you chose that number?

How can you be sure the verification fails because it's non-canonical and not just because it is a wrong public key and thus the verification fails anyway?

Reply:
It's a well-known non-canonical point, also used in other libraries: standards are full of constants and test vectors.

Reply:
Now TestGe25519IsCanonicalChecksNonCanonicals generates that non-canonical point and all the others.